### PR TITLE
Flash sim cleanup

### PR DESCRIPTION
**Pull request by kapi-no**
Resolves issues reported in the #8254

### CODE REVIEW

**Comment by kapi-no** [comment]
CC: @galak, @nvlsianpu

File: boards/x86/qemu_x86/Kconfig.defconfig [hunk 1]
```diff
@@ -9,6 +9,9 @@ config BUILD_OUTPUT_BIN
 config BOARD
 	default "qemu_x86"
 
+config FLASH_SIMULATOR
```
Comment (galak) [inline boards/x86/qemu_x86/Kconfig.defconfig]: should this be:

```
if FLASH

config FLASH_SIMULATOR
	default y

endif

Comment (kapi-no) [reply boards/x86/qemu_x86/Kconfig.defconfig]: yes, thank you, this is fixed now

**Review by galak — APPROVED** [review]

**Comment by kapi-no** [comment]
CI has passed.

@galak, could you merge it?

**Pull request merged**